src/Sentry/Internal/MainSentryEventProcessor.cs

```diff
@@ -136,9 +136,19 @@ public SentryEvent Process(SentryEvent @event)
                 @event.Release = _options.Release ?? Release;
             }
 
-            if (@event.Environment == null)
+            if (string.IsNullOrWhiteSpace(@event.Environment))
             {
-                @event.Environment = _options.Environment ?? EnvironmentLocator.Locate();
+                if (string.IsNullOrWhiteSpace(_options.Environment))
```

Comment:
I understand you did this right thing here to stay compatibel.
But could youb please change to use env var instead of options first? Values set via env var should take priority over the options.

Reply:
Totally. can do. 👍🏻

Reply:
Now refactored. Env var gets checked (and used) first, then options second, finally default value.